docs/proposal/20210311-single-controller-multitenancy.md

```diff
@@ -0,0 +1,255 @@
+---
+title: Single Controller Multitenancy
+authors:
+  - "@gab-satchi"
+reviewers:
+creation-date: 2021-03-11
+last-updated: 2021-03-11
+status: implementable
+see-also:
+  - https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/docs/proposal/20200506-single-controller-multitenancy.md
+  - https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/docs/proposals/20200720-single-controller-multitenancy.md  
+---
+
+# Single Controller Multitenancy
+
+## Table of Contents
+
+- [Title](#single-controller-multitenancy)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1---deploying-to-multiple-vcenters)
+      - [Story 2](#story-2---legacy-behaviour)
+    - [Requirements](#requirements)
+      - [Functional Requirements](#functional-requirements)
+        - [FR1](#fr1)
+        - [FR2](#fr2)
+        - [FR3](#fr3)
+        - [FR4](#fr4)
+      - [Non-Functional Requirements](#non-functional-requirements)
+        - [NFR1](#nfr1)
+        - [NFR2](#nfr2)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+        - [Current State](#current-state)
+        - [Proposed Changes](#proposed-changes)
+        - [Controller Changes](#controller-changes)
+        - [Clusterctl Changes](#clusterctl-changes)
+    - [Security Model](#security-model)
+        - [Roles](#roles)
+        - [RBAC](#rbac)
+        - [Namespace Restrictions](#namespace-restrictions)
+        - [CAPV Controller Requirements](#capv-controller-requirements)
+    - [Risks and Mitigations](#risks-and-mitigations)
+        - [Caching](#caching)
+  - [Alternatives](#alternatives)
+    - [Using Secrets](#using-only-secrets-to-specify-vsphere-accounts)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+- CAPV - An abbreviation of Cluster API Provider vSphere
+
+## Summary
+The CAPV controller is capable of managing infrastructure resources on a vCenter using the credentials it was provided during initialization. The credentials are provided via environment variables that get saved onto a secret that's used by the CAPV deployment. 
+
+Credentials provided are used for the entire lifetime of the CAPV deployment which means a CAPV cluster can become broken if the provisioning CAPV deployment were to be reconfigred for another set of credentials.
```

Comment:
How multitenancy can fix CAPV deployment credential issue? 
CAPA will start with credentials retrieved locally, then other roles could be used for creating workload clusters. 
When the local credentials (that CAPV uses) change, is the plan to update the secret?

Reply:
I don't think MT can fix any existing clusters from breaking when the deployment credentials are changed. This was more highlighting a cluster's dependance on the deployment credentials and if MT is fully utilized, it can move that dependency to a `VSphereAccount`